### PR TITLE
[Documentation] Replaced lerp dart docs with macro

### PR DESCRIPTION
**Pull request by johnsonmh**
* Replaced all instances of this comment block in `flutter/packages/flutter`. 
* Removed "The arguments must not be null" from the macro, since this changes depending on the class.

### CODE REVIEW

**Comment by dnfield** [comment]
LGTM

**Comment by HansMuller** [comment]
It seems odd that the macro template is in ThemeData but none of the other options seem more worthy.

LGTM too.

**Review by HansMuller — APPROVED** [review]

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -731,8 +731,6 @@ class ThemeData extends Diagnosticable {
   /// Linearly interpolate between two themes.
   ///
   /// {@template flutter.material.themeData.lerp}
-  /// The arguments must not be null.
-  ///
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: Why did you remove this statement? It's still true for this method.

Comment (Hixie) [reply packages/flutter/lib/src/material/theme_data.dart]: looks like it was just an oversight

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -731,8 +731,6 @@ class ThemeData extends Diagnosticable {
   /// Linearly interpolate between two themes.
   ///
   /// {@template flutter.material.themeData.lerp}
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: we should probably move this to the first inclusion of this text in the animation layer... it's weird that the animation layer refers to documentation in the material layer.

File: packages/flutter/lib/src/material/theme_data.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/theme_data.dart]: Yes sorry it was a mistake, I meant to move it outside of the template definition.  Will fix.

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/theme_data.dart]: I couldn't think of an obvious place, but you're right, it is weird to put it in `material/`.

Moved `painting/gradient.dart`

See: https://github.com/flutter/flutter/pull/22388/
